src/classifier/Pipe.cpp

```diff
@@ -443,15 +457,110 @@ void Pipe::Run() {
   LOG(INFO) << "Number of instances: " << num_instances;
   LOG(INFO) << "Time: " << diff_ms(end,start);
 
+  LOG(INFO) << "Cache size: " << parameters_->caching_weights_.size();
+  LOG(INFO) << "Cache hits: " << parameters_->caching_weights_.hits();
+  LOG(INFO) << "Cache misses: " << parameters_->caching_weights_.misses();
+
   if (options_->evaluate()) EndEvaluation();
 }
 
+//Variant of void Pipe::Run() , that launches threads for processing each instance
+void Pipe::RunWithThreads() {
```

Comment:
The architecture will be changed in the next version (and the multithread support will be removed from this context).